src/bindings/c/include/openvino/c/ov_model.h

```diff
@@ -83,20 +84,22 @@ OPENVINO_C_API(bool) ov_model_is_dynamic(const ov_model_t* model);
  * @param model A pointer to the ov_model_t.
  * @param tensor_names input tensor name (char *) list.
  * @param partialShape A PartialShape list.
- * @param cnt The item count in the list.
+ * @param size The item count in the list.
+ * @return Status code of the operation: OK(0) for success.
  */
 OPENVINO_C_API(ov_status_e)
 ov_model_reshape(const ov_model_t* model,
-                 const char* tensor_names[],
-                 const ov_partial_shape_t* partial_shapes[],
-                 size_t cnt);
+                 const char** tensor_names,
+                 const ov_partial_shape_t** partial_shapes,
+                 size_t size);
 
 /**
  * @brief Do reshape in model with partial shape for a specified name.
  * @ingroup model
  * @param model A pointer to the ov_model_t.
  * @param tensor_name input tensor name (char *).
  * @param partialShape A PartialShape.
+ * @return Status code of the operation: OK(0) for success.
  */
 OPENVINO_C_API(ov_status_e)
 ov_model_reshape_input_by_name(const ov_model_t* model,
```

Comment:
Looks like in current implementation we shouldn't pass PartialShape as a pointer to methods.

Reply:
why not? Do you mean pass a PartialShape  value?

Reply:
Yes, because PartialShape is not hidden struct any more.

Reply:
I think it is coding style issue, do you confirm we use "pass by value" to replace "pass by pointer"? If yes, maybe we will find that some code is "pass by value" and some code is "pass by pointer". It will be  untidy.

Reply:
We had the same discussion in big common PR. And looks like we pass by pointer only hidden structures.
Here looks like we can pass it by value, because now user works with value for such structures

Reply:
It is up to how to we define it, user will follow. But on the full picture, I still think keep the same behavior will be better.
@ilya-lavrenov  and @wangleis  how do you think about it?

Reply:
I propose to take a look to the our tests first of all :)
As I see in our tests on test side we don't work with pointer for the PartialShape at the current moment (example [1](https://github.com/openvinotoolkit/openvino/pull/12591/files#diff-9feab1b50258c82165bf228e42494f1b77cbdb8d3e633833b8e093f7068dc623R114), [2](https://github.com/openvinotoolkit/openvino/pull/12591/files#diff-4c47b52b3568c5b0c30298021801f07b170f5a330391c4b8b698c0035ef5e372R31), [3](https://github.com/openvinotoolkit/openvino/pull/12591/files#diff-4c11373907e8046e1b8257d56835b21bf2f781067e6360d5df7c9441c69dfd14R124) and etc.)

So it is why we should have API which allow to avoid any redundant symbols, for example:
```
ov_partial_shape shape;
ov_model_reshape_input_by_name(model, &shape, name); // with current implementation


ov_model_reshape_input_by_name(model, shape, name); // if we remove redundant pointers
```

Reply:
yes, in test case we apply local variable method to create partial_shape data and use its pointer, it is just to avoid allocating and free a buffer.

Reply:
But users also will use local variables, and in this case they don't need to have this additional pointer.

Reply:
I suppose that passing by pointer is a kind of optimization.. 

But I still don't have clear understanding which objects we should pass by pointer, which by value?
@riverlijunjie am I right that you want to pass by pointer to prevent copy by value?

Reply:
Actually, there are 2 points for "pass by pointer":
   1) align all methods to use pass by pointer
   2) some optimization by no data copying

Reply:
done - change to "pass by value" for such single C structure.